server/src/test/java/org/elasticsearch/gateway/GatewayIndexStateIT.java

```diff
@@ -0,0 +1,585 @@
+        assertThat(stateResponse.getState().metadata().index(tableName).getState(), equalTo(IndexMetadata.State.OPEN));
+        assertThat(stateResponse.getState().routingTable().index(tableName).shards().size(), equalTo(numPrimaries));
+        assertThat(stateResponse.getState().routingTable().index(tableName).shardsWithState(ShardRoutingState.STARTED).size(),
+            equalTo(totalNumShards));
+
+        logger.info("--> trying to get the indexed document on the first index");
+        execute("select id from test where id = 1");
+        assertThat(response.rowCount(), is(1L));
+
+
+        logger.info("--> closing test index...");
+        execute("alter table test close");
+
+        stateResponse = client().admin().cluster().prepareState().execute().actionGet(REQUEST_TIMEOUT);
+        assertThat(stateResponse.getState().metadata().index(tableName).getState(), equalTo(IndexMetadata.State.CLOSE));
+        assertThat(stateResponse.getState().routingTable().index(tableName), notNullValue());
+
+        logger.info("--> restarting nodes...");
+        internalCluster().fullRestart();
+        logger.info("--> waiting for two nodes and green status");
+        ensureGreen();
+
+        stateResponse = client().admin().cluster().prepareState().execute().actionGet(REQUEST_TIMEOUT);
+        assertThat(stateResponse.getState().metadata().index(tableName).getState(), equalTo(IndexMetadata.State.CLOSE));
+        assertThat(stateResponse.getState().routingTable().index(tableName), notNullValue());
+
+        logger.info("--> trying to index into a closed index ...");
+        try {
+            execute("insert into test (id) values (2)");
+            fail();
+        } catch (Exception e) {
+            // all is well
+        }
+
+        logger.info("--> opening index...");
+        execute("alter table test open");
+
+        logger.info("--> waiting for green status");
+        ensureGreen();
+
+        stateResponse = client().admin().cluster().prepareState().execute().actionGet();
+        assertThat(stateResponse.getState().metadata().index(tableName).getState(), equalTo(IndexMetadata.State.OPEN));
+        assertThat(stateResponse.getState().routingTable().index(tableName).shards().size(), equalTo(numPrimaries));
+        assertThat(stateResponse.getState().routingTable().index(tableName).shardsWithState(ShardRoutingState.STARTED).size(),
+            equalTo(totalNumShards));
+
+        logger.info("--> trying to get the indexed document on the first round (before close and shutdown)");
+        execute("select id from test where id = 1");
+        assertThat(response.rowCount(), is(1L));
+
+        logger.info("--> indexing a simple document");
+        execute("insert into test (id) values (2)");
+    }
+
+    @Test
+    @Ignore("Creating a table without any data node will take very long as internally at CrateDB, a table creation" +
```

Comment:
Could we use the slow annotation here instead, or does it currently fail?

Reply:
Was not sure if it's worth and maybe adds another flakiness. I've added a fixup.